secure.go

```diff
@@ -290,6 +295,10 @@ func (s *Secure) processRequest(w http.ResponseWriter, r *http.Request) (http.He
 	}
 
 	// Allowed hosts check.
+	if s.opt.AllowedHostsFunc != nil && !s.opt.IsDevelopment {
+		s.opt.AllowedHosts = append(s.opt.AllowedHosts, s.opt.AllowedHostsFunc()...)
+	}
+
```

Comment:
My suggested change will still call the function on every request. The way the code is written now will append the results of the function call into `s.opt.AllowedHosts` on every request... so that array will grow and grow and grow! Instead of appending to `s.opt.AllowedHosts`, we can create a new variable that combines both the static list and the function call results. Then on lines 302 and 312 we can update the references from `s.opt.AllowedHosts` to the new variable we created (I called it `combinedAllowedHosts` in my suggestion above)

Reply:
Am actually wondering if we can modify this to pass in the request to the function, what do you think? Or have another function` isOriginAllowed(host string) `

Reply:
I like the idea of passing the request in! I'll try to prototype that soon and see how it fits in.